### PR TITLE
Bump k8s libraries to v0.30.0

### PR DESCRIPTION
**Pull request by Neo2308**
**Description of the change:**
* Bump k8s libraries to v0.30.0
* Bump controller-runtime to v0.18.2

**Motivation for the change:**
Relates to: https://github.com/operator-framework/operator-sdk/issues/6730

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -23,7 +25,7 @@ require (
 	github.com/onsi/gomega v1.33.0
 	github.com/opencontainers/go-digest v1.0.0
 	github.com/opencontainers/image-spec v1.1.0
-	github.com/operator-framework/api v0.23.0
+	github.com/operator-framework/api v0.23.1-0.20240429145053-7d638a091922
```
Comment (perdasilva) [inline go.mod:28]: I'm cutting a release of the api so we can update this reference to v0.24.0

Comment (perdasilva) [reply go.mod:28]: done!! =D https://github.com/operator-framework/api/releases/tag/v0.24.0

Comment (Neo2308) [reply go.mod]: Done :)

**Comment by perdasilva** [comment]
@Neo2308 could you please bump controller-runtime to v0.18.2 so we can close #1305

File: go.mod [hunk 2]
```diff
@@ -41,12 +43,12 @@ require (
...
-	k8s.io/apimachinery v0.29.3
-	k8s.io/client-go v0.29.3
-	k8s.io/kubectl v0.28.5
-	sigs.k8s.io/controller-runtime v0.17.3
+	k8s.io/api v0.30.0
+	k8s.io/apiextensions-apiserver v0.30.0
+	k8s.io/apimachinery v0.30.0
+	k8s.io/client-go v0.30.0
+	k8s.io/kubectl v0.30.0
+	sigs.k8s.io/controller-runtime v0.18.1
```
Comment (everettraven) [inline go.mod:51]:
```suggestion
	sigs.k8s.io/controller-runtime v0.18.2
```

Comment (Neo2308) [reply go.mod]: Done :)

**Comment by Neo2308** [comment]
@perdasilva / @everettraven can you review this PR now?

**Comment by perdasilva** [comment]
@grokspawn can we merge this one? Should be ok, right?

**Review by everettraven — APPROVED** [review]
/lgtm

**Review by grokspawn — APPROVED** [review]

**Pull request merged**